### PR TITLE
Distro cleanup

### PR DESCRIPTION
**Pull request by teg**
After the recent addition of new interfaces to the distro package, drop dead code and do other mechanical cleanups.

### CODE REVIEW

File: internal/distro/fedora31/distro.go [hunk 1]
```diff
@@ -16,45 +15,45 @@ import (
...
+	excludedPackages []string
+	enabledServices  []string
+	disabledServices []string
+	kernelOptions    string
+	bootable         bool
+	defaultSize      uint64
+	assembler        func(uefi bool, size uint64) *osbuild.Assembler
+}
+
+type fedora31Arch struct {
```
Comment (msehnout) [inline internal/distro/fedora31/distro.go]: What is the reason for having 2 private "arch" struct here? (`fedora31Arch` and `arch`), seems to me that they have name in common, so the only difference is a pointer to the distro struct.

File: internal/distro/fedora31/distro_test.go [hunk 2]
```diff
@@ -1,31 +1,11 @@
 package fedora31_test
 
 import (
-	"reflect"
 	"testing"
 
 	"github.com/osbuild/osbuild-composer/internal/distro/fedora31"
 )
 
-func TestListOutputFormats(t *testing.T) {
```
Comment (msehnout) [inline internal/distro/fedora31/distro_test.go]: This should still be possible, just more complicated, no? For example a test case testing a matrix of arches and image types.

File: internal/distro/fedora31/distro.go (hunk 1 above)
Comment (teg) [reply internal/distro/fedora31/distro.go]: This was to make the (already very large) patch a bit smaller. Would be a natural next cleanup. You are right, this is clearly not how it should be.

File: internal/distro/fedora31/distro_test.go (hunk 2 above)
Comment (teg) [reply internal/distro/fedora31/distro_test.go]: Yeah, totally. Also need tests for the other newly introduced functions. I'll post a follow up to add this.

File: internal/distro/fedora30/distro_test.go [hunk 3]
```diff
@@ -1,31 +1,11 @@
 package fedora30_test
 
 import (
-	"reflect"
 	"testing"
 
 	"github.com/osbuild/osbuild-composer/internal/distro/fedora30"
 )
 
-func TestListOutputFormats(t *testing.T) {
```
Comment (larskarlitski) [inline internal/distro/fedora30/distro_test.go]: Why did you drop these?

File: internal/distro/fedora31/distro.go (hunk 1 above)
Comment (larskarlitski) [reply internal/distro/fedora31/distro.go]: We discussed this on the last PR: https://github.com/osbuild/osbuild-composer/pull/375#discussion_r396374144

Tom's idea is to do this in chunks.

File: internal/distro/fedora30/distro_test.go (hunk 3 above)
Comment (larskarlitski) [reply internal/distro/fedora30/distro_test.go]: Sorry, didn't see that you already discussed this with Martin above.

**Pull request merged**